v1/ao/internal/utils/status.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2017 Librato, Inc. All rights reserved.
+
+package utils
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+)
+
+var (
+	installDir        string
+	installTsInSec    int64
+	lastRestartInUSec int64
+)
+
+func init() {
+	installDir = initInstallDir()
+	installTsInSec = initInstallTsInSec()
+	lastRestartInUSec = initLastRestartInUSec()
+}
+
+func initInstallDir() string {
+	_, path, _, ok := runtime.Caller(0)
+	if !ok {
+		return "unknown"
+	}
+
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return "unknown"
+	}
+
+	prevPath := string(os.PathSeparator)
+	for path != prevPath {
+		base := filepath.Base(path)
+		if base == "ao" {
```

Comment:
Is it guaranteed that the user would install to a location with "ao" as a part of it?

And also what would runtime.Caller(0) returns in the `path`, can you please give an example of the value returned right now :)

Reply:
The `path` returned by `runtime.Caller(0)` is the path of the file where this function (`runtime.Caller(0)`) is called. e.g., it returns `/Users/yang/go/src/github.com/appoptics/appoptics-apm-go/v1/ao/internal/utils/status.go` in this call. 

The `ao` is part of the path because it's the package name.